lib/registration_util.dart

```diff
@@ -11,7 +10,11 @@ class RegistrationUtil{
   /// ...the password contains less than 2 digits
   /// ...the confirmed password is not the same as the password
 
-  bool validateRegistrationUserInput(String username, String password, String confirmedPassword){
+  bool validateRegistrationUserInput(
+      String username, String password, String confirmedPassword) {
+    if (username.isEmpty) return false;
```

Comment:
A nghĩ nên dùng isBlank thay cho isEmpty.
// password contains less than 2 digits: What is a digit?